freud/box.pyx

```diff
@@ -921,6 +942,43 @@ cdef class Box:
                             "positional argument: L")
         return cls(Lx=L, Ly=L, Lz=0, xy=0, xz=0, yz=0, is2D=True)
 
+    @classmethod
+    def from_box_lengths_and_angles(
+        cls, L1, L2, L3, alpha, beta, gamma, dimensions=None,
+    ):
+        r"""Construct a box from lengths and angles.
+
+        All the angles provided must be between 0 and :math:`\pi`.
```

Comment:
This matches the [cif spec](https://www.iucr.org/__data/iucr/cifdic_html/1/cif_core.dic/Icell_angle_.html) - nice.